src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateItemEventHandler.java

```diff
@@ -394,33 +397,37 @@ private Future<Item> updateItemAndRetryIfOLExists(Item item, ItemCollection item
     return promise.future();
   }
 
-  private void processOLError(DataImportEventPayload dataImportEventPayload, CompletableFuture<DataImportEventPayload> future, ItemCollection itemCollection, Item item, List<PartialError> errors) {
+  private void processOLError(DataImportEventPayload dataImportEventPayload, CompletableFuture<DataImportEventPayload> future, ItemCollection itemCollection, List<Item> expiredItems, List<PartialError> errors) {
     int currentRetryNumber = dataImportEventPayload.getContext().get(CURRENT_RETRY_NUMBER) == null ? 0 : Integer.parseInt(dataImportEventPayload.getContext().get(CURRENT_RETRY_NUMBER));
+
+
     if (currentRetryNumber < MAX_RETRIES_COUNT) {
       dataImportEventPayload.getContext().put(CURRENT_RETRY_NUMBER, String.valueOf(currentRetryNumber + 1));
-      LOGGER.warn("processOLError:: Error updating Item by id '{}'. Retry UpdateItemEventHandler handler...", item.getId());
-      itemCollection.findById(item.getId())
-        .thenAccept(actuaItem -> prepareDataAndReInvokeCurrentHandler(dataImportEventPayload, future, actuaItem))
-        .thenAccept(v -> dataImportEventPayload.getContext().remove(CURRENT_RETRY_NUMBER))
-        .exceptionally(e -> {
+      LOGGER.warn("processOLError:: Error updating Items. Expired Items: '{} '.Current retry number = '{}'. Retry UpdateItemEventHandler handler...", expiredItems, currentRetryNumber);
+      getActualItemsList(expiredItems, itemCollection)
+        .onSuccess(actualItemsList -> prepareDataAndReInvokeCurrentHandler(dataImportEventPayload, future, actualItemsList, errors))
+        .onFailure(e -> {
+          String errMessage = format("Cannot get actual Items.Expired Items: '%s' for jobExecutionId '%s'. Error: %s ", expiredItems, dataImportEventPayload.getJobExecutionId(), e.getCause());
+          for (Item expiredHolding : expiredItems) {
```

Comment:
Looks like there should be expiredItem instead of expiredHolding here and in the loop below

Reply:
@tatsiana-tarhonskaya  Thanks. It is a typo.